engine.js

```diff
@@ -473,24 +484,34 @@ var handleUniqueLoanNumberErrors = function(counts) {
 
     /* hmda-syntactical */
     HMDAEngine.isValidControlNumber = function(hmdaFile) {
-        return true;
+        return readResponseSync(HMDAEngine.getAPIURL(), 'isValidControlNumber', HMDAEngine.getRuleYear(), 
+            [hmdaFile.transmittalSheet.agencyCode, hmdaFile.transmittalSheet.respondentID]);
     };
 
     /* lar-validity */
     HMDAEngine.isValidMetroArea = function(metroArea) {
-        return true;
+        return readResponseSync(HMDAEngine.getAPIURL(), 'isValidMSA', HMDAEngine.getRuleYear(), [metroArea]);
     };
 
     HMDAEngine.isValidMsaMdStateAndCountyCombo = function(metroArea, fipsState, fipsCounty) {
-        return true;
+        return readResponseSync(HMDAEngine.getAPIURL(), 'isValidMSAStateCounty', HMDAEngine.getRuleYear(), [metroArea, fipsState, fipsCounty]);
     };
 
     HMDAEngine.isValidStateAndCounty = function(fipsState, fipsCounty) {
-        return true;
+        return readResponseSync(HMDAEngine.getAPIURL(), 'isValidStateCounty', HMDAEngine.getRuleYear(), [fipsState, fipsCounty]);
     };
 
     HMDAEngine.isValidCensusTractCombo = function(censusTract, metroArea, fipsState, fipsCounty) {
```

Comment:
@doelleri The logic on the API side for isValidCensusCombination would only return true for tract = 'NA' when the small county flag is enabled for that county. There are two different API calls that check the census collection with state and county and no tract and because the choice of return logic in isValidCensusCombination is decided based on the inputs there's no way to separate the logic for this call from the other one without adding a new input to the function.